packages/flutter/lib/src/material/tabs.dart

```diff
@@ -871,20 +876,32 @@ class _TabBarState extends State<TabBar> {
     _controller.animateTo(index);
   }
 
-  Widget _buildStyledTab(Widget child, bool selected, Animation<double> animation) {
+  Widget _buildStyledTab(Widget child, bool selected, Animation<double> animation, ThemeData themeData) {
```

Comment:
Good call, it makes much more sense to move down. Rather than put it in the `CustomPaint` constructor call, I moved it down into the `_TabStyle`'s `build` method. This build method already retrieves the `themeData` from the context, and putting it here lets me only make the decision in one place, which I think makes the most sense. WDYT?

Reply:
Agreed!